src/hooks/resources/device.ts

```diff
@@ -576,3 +589,59 @@ async function checkSupervisorReleaseUpgrades(
 		}
 	}
 }
+
+async function setSupervisorReleaseResource(
+	api: sbvrUtils.PinejsClient,
+	deviceIds: number[],
+	supervisorVersion: string,
+) {
+	// This function _should_ get called with one device (since each supervisor reports itself)
+	const device = (await api.get({
+		resource: 'device',
+		id: deviceIds[0],
+		options: {
+			$expand: ['is_of__device_type'],
+			$select: ['id', 'is_of__device_type', 'supervisor_version'],
+		},
+	})) as AnyObject;
+
+	if (device.supervisor_version != null) {
+		// if the device already has a supervisor_version, just bail.
+		return null;
+	}
+
+	const supervisorRelease = (await api.get({
+		resource: 'supervisor_release',
+		options: {
+			$select: ['id'],
+			$filter: {
+				supervisor_version: `v${supervisorVersion}`,
+				is_for__device_type: {
+					$any: {
+						$alias: 'dt',
+						$expr: {
+							dt: {
+								id: device.is_of__device_type[0].id,
+							},
+						},
+					},
+				},
+			},
+		},
+	})) as AnyObject[];
+
+	if (supervisorRelease != null) {
+		return await api.patch({
```

Comment:
The return + await is unnecessary here:
```suggestion
		return api.patch({
```
or
```suggestion
		await api.patch({
```

Reply:
done